### ISSUE TITLE
Number.isNaN support

### ISSUE THREAD

**Issue by ebourg** [polyfillpolyfill/polyfill-service#522]
Hi,

Could you consider adding a polyfill for the ES6 function `Number.isNaN()` please? This function is supported by all major browsers but IE, this is documented in MDN:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN

```
Number.isNaN = Number.isNaN || function(value) {
    return typeof value === "number" && isNaN(value);
}
```

Thank you

**Comment by triblondon** [comment]
We'd certainly welcome a submission that adds this polyfill.

**Comment by Ikkyu-os** [comment]
If it's not urgent, i'd be willing to implement this one, to learn a bit how polyfill-service is working.

**Comment by triblondon** [comment]
@Ikkyu-os well volunteered :-)  We really appreciate your help.

Contributor guide is here: http://cdn.polyfill.io/v2/docs/contributing

**Comment by Ikkyu-os** [comment]
FYI, I started working on this issue (at least doing some research pending https://github.com/Financial-Times/polyfill-service/issues/547).

Worst case, I'll find my old linux box to move on.

**Comment by triblondon** [comment]
@Ikkyu-os are you still working on this?

**Comment by Ikkyu-os** [comment]
Hi.

Sorry, was pretty busy with work. I had a first version locally, but then I found out that test suite is also not working well on Windows OS :) 
So I put it on hold, but I'm still willing to fix it. I would actually need to fix tests execution on windows first.

**Comment by Ikkyu-os** [comment]
Hi @triblondon 

I implemented a first version of the polyfill, code is available on my fork: https://github.com/Ikkyu-os/polyfill-service/tree/numberIsNaN/

However, I have difficulties testing it with my test suite (not yet final) with the mocha test suite, using same url pattern as proposed in the doc
http://127.0.0.1:3000/test/tests?feature=Number.isNaN&mode=targeted (or mode=all)

Whatever the browser (last FF, FF 11.0, IE 11.0, IE 8.0 on VM), the test always load empty with the following status in the page:

```
passes: 0 failures: 0 duration: 0s
```

Indeed, the polyfill is not returned by http://127.0.0.1:3000/v2/polyfill.js?flags=gated,always&features=   HTTP request.
I totally understand I'm missing something, which could likely be that my polyfill is not properly integrated in my local build
My build steps:
- grunt build  (here I can see my Number/isNaN properly listed in the build)
- bin/polyfill-service   (had to start manually as 'grunt dev' can't work on windows system due to .pid  not recognize as a command)

Any hint on what I am missing would be appreciated. I'm pretty sure what is missing sounds dumb, but I still need to familiarize with the build system :)

**Comment by triblondon** [comment]
This happens when you combine `targeted` test mode with a polyfill that is not targeted to the current browser.  Try either:
- `mode=all`; or
- Update your config.json to target more browsers

It looks from your config like it ought to be serving to IE, so I don't know why that's not happening.  Is your IE VM definitely being recognised as ie/8?

Finally, we seem to often have issues with the `<=` semver operator. We encourage use of `<` or `-` instead.

**Comment by Ikkyu-os** [comment]
Ok.
I really think I don't deploy my polyfill correctly with grunt build..
Could be due to my setup again :)

Here's how the polyfill/__dist is built locally:
|---- Array/ 
|---- Number/
|------- isNan.json
|- JSON.json
|- matchMedia.json
etc...

If I move my polyfill at the root of __dist/ (and not in __dist/Number/ ) and rename the file Number.isNaN.json the polyfill is properly served and tests are executed in the browser.

I tested with an example from the contributing guide : http://127.0.0.1:3000/test/tests?feature=Array.from&mode=targeted  (which polyfill is also built in subdirectories), and same issue occurs.
For yet an unknown reason, the service doesn't serve the polyfills in subdirectories.

At least, I can continue to work on the polyfill itself, and will add more tests.

**Comment by triblondon** [comment]
Hm, that's weird. I think your earlier issues were related to directory creation as well, right? So it's possible that this is a Windows thing.  Your `__dist` directory should build flat:

![image](https://cloud.githubusercontent.com/assets/1735391/13034610/e79ee1ec-d331-11e5-9cd0-29ce7de99dbb.png)

**Comment by Ikkyu-os** [comment]
And indeed, it's likely a windows issue :)
Here's how my __dist is built:
![polyfilldistdirectory](https://cloud.githubusercontent.com/assets/695806/13034642/271f74e0-d33b-11e5-8f89-7264a07f30b3.png)

I'll have to open another issue for this, although it will be harder than the previous issue, and it fails silently this time.

Thanks for your help anyway, I will do a PR for this polyfill today.

**Comment by JakeChampion** [comment]
Closing as work was merged in #608

**Comment by ebourg** [comment]
Thanks!
